tests/common/aws_test_runner.c

```diff
@@ -278,13 +388,20 @@ void TEST_RUNNER_RunTests_task( void * pvParameters )
     UnityFixture.NameFilter = testrunnerTEST_FILTER;
     UnityFixture.RepeatCount = 1;
 
-    /* Add sufficient delay before starting tests on device to allow
-     * device to be available as serial port connection to the host machine
-     * OS.
-     * The serial console is used by host machine to view device logs. */
-    vTaskDelay( pdMS_TO_TICKS( AWS_TEST_RUNNER_DELAY_MS ) );
+    #if defined( AWS_TEST_RUNNER_ENABLE_CLI_INPUT ) && ( AWS_TEST_RUNNER_ENABLE_CLI_INPUT == 1 )
+        prvTestRunnerConsole();
+    #else
+
+        /* Add sufficient delay before starting tests on device to allow
+         * device to be available as serial port connection to the host machine
+         * OS.
+         * The serial console is used by host machine to view device logs. */
+        vTaskDelay( pdMS_TO_TICKS( AWS_TEST_RUNNER_DELAY_MS ) );
 
-    UNITY_BEGIN();
+        UNITY_BEGIN();
+
+        RunTests();
```

Comment:
It seems the memory leak test wont start before the test begin
`testrunnerFULL_MEMORYLEAK_ENABLED `

Reply:
Fixed by moving all the test setup, execution of tests and teardown logic in common function.